mypy_django_plugin/transformers/fields.py

```diff
@@ -201,6 +205,156 @@ def drop_combinable(_type: MypyType) -> Optional[MypyType]:
     return helpers.reparametrize_instance(default_return_type, args)
 
 
+class M2MThrough(NamedTuple):
+    arg: Optional[Expression]
+    model: ProperType
+
+
+class M2MTo(NamedTuple):
+    arg: Expression
+    model: ProperType
+    self: bool  # ManyToManyField('self', ...)
+
+
+class M2MArguments(NamedTuple):
+    to: M2MTo
+    through: Optional[M2MThrough]
+
+
+def get_model_from_expression(
+    expr: Expression,
+    *,
+    api: Union[TypeChecker, SemanticAnalyzer],
+    django_context: DjangoContext,
+) -> Optional[ProperType]:
+    """
+    Attempts to resolve an expression to a 'TypeInfo' instance. Any lazy reference
+    argument(e.g. "<app_label>.<object_name>") to a Django model is also attempted.
+    """
+    # TODO: Handle settings.AUTH_USER_MODEL?
+    if isinstance(expr, NameExpr) and isinstance(expr.node, TypeInfo):
+        if (
+            expr.node.metaclass_type is not None
+            and expr.node.metaclass_type.type.fullname == fullnames.MODEL_METACLASS_FULLNAME
+        ):
```

Comment:
I suppose this could be broken out to a type guard helper, if we want?

```python
ModelTypeInfo = NewType("ModelTypeInfo", TypeInfo)
def is_model_info(info: TypeInfo) -> TypeGuard[ModelTypeInfo]:
    return (
        info.metaclass_type is not None
        and info.metaclass_type.type.fullname == fullnames.MODEL_METACLASS_FULLNAME
    )
```